truffle/src/main/java/org/jruby/truffle/nodes/core/KernelNodes.java

```diff
@@ -1553,9 +1553,21 @@ public RubyNode coerceFeatureToPath(RubyNode feature) {
 
         @Specialization(guards = "isRubyString(featureString)")
         public boolean require(DynamicObject featureString) {
-            CompilerDirectives.transferToInterpreter();
             final String feature = featureString.toString();
 
+            // Pysch loads either the jar or the so - we need to intercept
+
+            if (feature.equals("psych.so") && RubyCallStack.getCallerFrame(getContext()).getCallNode()
```

Comment:
In the past, we've handled these one-offs in `FeatureLoader`. I don't think we have any currently so maybe it's not a problem. I just want to make sure we all know where to look for these sorts of things in the future.